### PR TITLE
chore: Bump `@metamask/transaction-controller` to 25.1.0 (along with related packages)

### PR DESCRIPTION
**Pull request by mcmire**
## **Description**

This PR upgrades `@metamask/transaction-controller` from 23.1.0 to 25.1.0. This version includes several major changes, but the one that is most relevant for this PR is that TransactionController now inherits from BaseController v2. For more, see the [changelog](https://github.com/MetaMask/core/blob/%40metamask/transaction-controller%4025.1.0/packages/transaction-controller/CHANGELOG.md).

In order to satisfy peer dependency constraints, the following packages also needed to be bumped. There aren't any functional changes to these packages:

- `@metamask/approval-controller` 5.1.0 -> 6.0.0 ([view changelog](https://github.com/MetaMask/core/blob/%40metamask/approval-controller%406.0.0/packages/approval-controller/CHANGELOG.md))
- `@metamask/gas-fee-controller` 13.0.0 -> 14.0.0 ([view changelog](https://github.com/MetaMask/core/blob/%40metamask/gas-fee-controller%4014.0.0/packages/gas-fee-controller/CHANGELOG.md))
- `@metamask/network-controller` 17.2.0 -> 18.0.0 ([view changelog](https://github.com/MetaMask/core/blob/%40metamask/network-controller%4018.0.0/packages/network-controller/CHANGELOG.md))
- `@metamask/user-operation-controller` 4.0.0 -> 6.0.0 ([view changelog](https://github.com/MetaMask/core/blob/%40metamask/user-operation-controller%406.0.0/packages/user-operation-controller/CHANGELOG.md))
- `@babel/runtime` 7.23.2 -> 7.24.0 (the patch file has also been regenerated)

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23252?quickstart=1)

## **Related issues**

PR that upgraded TransactionController to BaseController v2: https://github.com/MetaMask/core/issues/3758

## **Manual testing steps**

We'll want to verify that features related to transactions still work. Sending a transaction,  seeing it confirmed, etc.

## **Screenshots/Recordings**

(No screenshots to show)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by mcmire** [comment]
@metamaskbot update-policies

**Comment by mcmire** [comment]
@metamaskbot update-policies

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -1480,86 +1480,83 @@ export default class MetamaskController extends EventEmitter {
...
+        ),
+      getNetworkState: () => this.networkController.state,
+      getPermittedAccounts: this.getPermittedAccounts.bind(this),
+      getSavedGasFees: () =>
+        this.preferencesController.store.getState().advancedGasFee[
+          this.networkController.state.providerConfig.chainId
+        ],
+      getSelectedAddress: () =>
+        this.accountsController.getSelectedAccount().address,
+      isMultichainEnabled: process.env.TRANSACTION_MULTICHAIN,
```
Comment (matthewwalsh0) [inline app/scripts/metamask-controller.js:1519]: Apologies for the overlap, but assuming this PR will adopt the next `TransactionController` release, please can we also add: `isSimulationEnabled: () => false` to disable simulation until we've introduced the associated UX, thanks!

Comment (mcmire) [reply app/scripts/metamask-controller.js]: Added in [4e709ae](https://github.com/MetaMask/metamask-extension/pull/23252/commits/4e709ae7363acfdff220601723a1a120b85c0f8f).

**Comment by mcmire** [comment]
@metamaskbot update-policies

**Comment by mcmire** [comment]
Judging by recent builds, the e2e failures seem like known flaky tests and therefore seem unrelated to the package bumps.

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -1502,86 +1502,84 @@ export default class MetamaskController extends EventEmitter {
...
+        allowedActions: [
+          `${this.approvalController.name}:addRequest`,
+          'NetworkController:findNetworkClientIdByChainId',
+          'NetworkController:getNetworkClientById',
+        ],
+        allowedEvents: [`NetworkController:stateChange`],
+      });
+    this.txController = new TransactionController({
+      blockTracker: this.blockTracker,
+      cancelMultiplier: 1.1,
```
Comment (matthewwalsh0) [inline app/scripts/metamask-controller.js:1517]: Minor, but this property no longer exists.

File: app/scripts/metamask-controller.js [hunk 3]
```diff
@@ -1502,86 +1502,84 @@ export default class MetamaskController extends EventEmitter {
...
-          getAdditionalSignArguments: getAdditionalSignArgumentsMMI.bind(this),
-          ///: END:ONLY_INCLUDE_IF
-        },
+        queryEntireHistory: false,
+        updateTransactions: false,
       },
-      {
-        sign: (...args) => this.keyringController.signTransaction(...args),
+      isMultichainEnabled: process.env.TRANSACTION_MULTICHAIN,
+      isSimulationEnabled: () => false,
```
Comment (matthewwalsh0) [inline app/scripts/metamask-controller.js:1554]: ❤️

**Review by matthewwalsh0 — DISMISSED** [review]

File: app/scripts/metamask-controller.js (hunk 2 above)
Comment (mcmire) [reply app/scripts/metamask-controller.js]: Good point! Removed in [668137d](https://github.com/MetaMask/metamask-extension/pull/23252/commits/668137dc2a9b3c2e5c5676a2f2047d69ebdc5e94).

**Review by OGPoyraz — DISMISSED** [review]

**Comment by matthewwalsh0** [comment]
@metamaskbot update-policies

**Review by matthewwalsh0 — APPROVED** [review]

**Review by legobeat — APPROVED** [review]

**Pull request merged**